cudaaligner/src/aligner_global.cpp

```diff
@@ -51,14 +51,15 @@ AlignerGlobal::AlignerGlobal(int32_t max_query_length, int32_t max_target_length
         throw std::runtime_error("Max alignments must be at least 1.");
     }
     scoped_device_switch dev(device_id);
-    sequences_d_        = device_buffer<char>(sequences_h_.size());
-    sequence_lengths_d_ = device_buffer<int32_t>(sequence_lengths_h_.size());
-    results_d_          = device_buffer<int8_t>(results_h_.size());
-    result_lengths_d_   = device_buffer<int32_t>(result_lengths_h_.size());
-    device_memset_async(sequences_d_, 0, stream);
-    device_memset_async(sequence_lengths_d_, 0, stream);
-    device_memset_async(results_d_, 0, stream);
-    device_memset_async(result_lengths_d_, 0, stream);
+    sequences_d_        = device_buffer<char>(sequences_h_.size(), allocator, stream);
+    sequence_lengths_d_ = device_buffer<int32_t>(sequence_lengths_h_.size(), allocator, stream);
+    results_d_          = device_buffer<int8_t>(results_h_.size(), allocator, stream);
+    result_lengths_d_   = device_buffer<int32_t>(result_lengths_h_.size(), allocator, stream);
+
+    CGA_CU_CHECK_ERR(cudaMemsetAsync(sequences_d_.data(), 0, sizeof(char) * sequences_d_.size(), stream));
```

Comment:
Can this be done with `device_copy_n`?

Reply:
I think we don't have function to provide `cudaMemset` at the moment. Will leave it as it is.

Reply:
can I resolve this?

Reply:
True, we can either set a single value of copy an array. Yes, leave it as it is